src/Features/TerrainBlending.cpp

```diff
@@ -59,43 +59,43 @@ void TerrainBlending::PostPostLoad()
 
 void TerrainBlending::SetupGeometry(RE::BSRenderPass*)
 {
-	if (!enableBlending) {
-		Bindings::GetSingleton()->SetOverwriteTerrainMode(false);
-		Bindings::GetSingleton()->SetOverwriteTerrainMaskingMode(Bindings::TerrainMaskMode::kNone);
-		return;
-	}
-
-	bool validPass = TerrainBlending::ValidBlendingPass(a_pass);
-
-	if (a_pass->shaderProperty->flags.all(RE::BSShaderProperty::EShaderPropertyFlag::kMultiTextureLandscape)) {
-		if (validPass) {
-			Bindings::GetSingleton()->SetOverwriteTerrainMode(true);
-			Bindings::GetSingleton()->SetOverwriteTerrainMaskingMode(Bindings::TerrainMaskMode::kRead);
-
-			auto& context = State::GetSingleton()->context;
-			auto view = Bindings::GetSingleton()->terrainBlendingMask ? Bindings::GetSingleton()->terrainBlendingMask->srv.get() : nullptr;
-			if (view)
-				context->PSSetShaderResources(35, 1, &view);
-		} else {
-			Bindings::GetSingleton()->SetOverwriteTerrainMode(false);
-			Bindings::GetSingleton()->SetOverwriteTerrainMaskingMode(Bindings::TerrainMaskMode::kNone);
-		}
-	} else {
-		Bindings::GetSingleton()->SetOverwriteTerrainMode(false);
-		bool staticReference = false;
-		if (validPass) {
-			if (auto ref = a_pass->geometry->GetUserData()) {
-				if (auto base = ref->GetBaseObject()) {
-					if (base->As<RE::TESObjectSTAT>()) {
-						staticReference = true;
-					}
-				}
-			}
-		}
-		Bindings::GetSingleton()->SetOverwriteTerrainMaskingMode(validPass && !staticReference ?
-																	 (!REL::Module::IsVR() ?
-																			 Bindings::TerrainMaskMode::kWrite :
-																			 Bindings::TerrainMaskMode::kRead) :  // Fix VR artifact where static objects would appear shifted in each eye
-																	 Bindings::TerrainMaskMode::kNone);
-	}
+	//if (!enableBlending) {
+	//	Bindings::GetSingleton()->SetOverwriteTerrainMode(false);
+	//	Bindings::GetSingleton()->SetOverwriteTerrainMaskingMode(Bindings::TerrainMaskMode::kNone);
+	//	return;
+	//}
+
+	//bool validPass = TerrainBlending::ValidBlendingPass(a_pass);
+
+	//if (a_pass->shaderProperty->flags.all(RE::BSShaderProperty::EShaderPropertyFlag::kMultiTextureLandscape)) {
+	//	if (validPass) {
+	//		Bindings::GetSingleton()->SetOverwriteTerrainMode(true);
+	//		Bindings::GetSingleton()->SetOverwriteTerrainMaskingMode(Bindings::TerrainMaskMode::kRead);
+
+	//		auto context = RE::BSGraphics::Renderer::GetSingleton()->GetRuntimeData().context;
```

Comment:
Maybe fix this comment to `auto& context = State::GetSingleton()->context;` to make it work in VR when this is uncommented?

Reply:
This will be reimplemented later